src/ArraySetops.chpl

```diff
@@ -0,0 +1,135 @@
+/* Array set operations
+ includes intersection, union, xor, and diff
+
+ currently, only performs operations with integer arrays 
+ */
+
+module ArraySetops
+{
+    use ServerConfig;
+
+    use SymArrayDmap;
+
+    use RadixSortLSD;
+    use Unique;
+    use Indexing;
+    use In1d;
+
+
+    /*
+    Small bound const. Brute force in1d implementation recommended.
+    */
+    private config const sBound = 2**4; 
+
+    /*
+    Medium bound const. Per locale associative domain in1d implementation recommended.
+    */
+    private config const mBound = 2**25; 
+
+    // returns intersection of 2 arrays
+    proc intersect1d(a: [] int, b: [] int, assume_unique: bool) {
+      //if not unique, unique sort arrays then perform operation
+      if (!assume_unique) {
+        var a1  = uniqueSort(a, false);
+        var b1  = uniqueSort(b, false);
+        return intersect1dHelper(a1, b1);
+      }
+      return intersect1dHelper(a,b);
+    }
+
+    // Get intersect of 2 arrays
+    // first concatenates the 2 arrays, then
+    // sorts arrays and removes all values that
+    // only occur once
+    proc intersect1dHelper(a: [] ?t, b: [] t) {
+      var aux = radixSortLSD_keys(concatset(a,b));
+
+      var head = sliceEnd(aux);
+      var mask = head == tail(aux);
```

Comment:
It seems that we are making a copy of the two sections of the array instead of doing the simple comaprasion between array elements which would avoid the copy all together.
Could we have the head/tail function just compute the slice and return it, or maybe just do something like...
```chapel
var mask = aux[aux.domain.low()..(aux.domain.high()-1)] == aux[(aux.domain.low()+1)..aux.domain.high()];
```
@bradcray what do you think?

Reply:
Yeah, I agree that this feels pretty heavyweight since `head` and `tail` represent all of the array elements but one.  I suspect/hope we could get better performance by doing something like:

```chapel
const maskInds = aux.domain#(aux.size-1);  // maskInds describes all indices but the last
var mask = [i in maskInds] aux[i] == aux[i+1];
```

The main overheads of this implementation would be:

(a) creating a new distributed domain, `maskInds`, but this should be strictly less expensive than the current slicing-based approach, since it'll create distributed domains and arrays.
(b) the overheads for accessing potentially-remote data in the `aux[i+1]` expression
(c) and potentially similar overheads for the `aux[i]` expression unless @e-kayrakli's ongoing work on optimizing aligned accesses could handle it automatically... assuming not, `aux.localAccess[i]` would take care of it, so is probably where we should start).

We could work around (b) with more effort if it turned out to be overly expensive (for example, we could have per-locale loops that iterated over the first local_n-1 iterations using `.localAccess`, stripping off the final iteration that is guaranteed to be remote).  But my guess is that even without those changes, this version would still be a net win over creating the two distributed n-1 element arrays.

Reply:
> and potentially similar overheads for the aux[i] expression unless @e-kayrakli's ongoing work on optimizing aligned accesses could handle it automatically... assuming not, aux.localAccess[i] would take care of it, so is probably where we should start)

I expect that optimization to not fire for this pattern today. But I'll get to it soon. I am confident that it'll fire in the next release.